packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts

```diff
@@ -179,3 +182,23 @@ function addAlgoliaAgents(
     searchClient.addAlgoliaAgent!(userAgent!);
   });
 }
+
+function warnNextRouter<TUiState extends UiState, TRouteState>(
+  routing: UseInstantSearchApiProps<TUiState, TRouteState>['routing']
+) {
+  if (!routing || typeof window === 'undefined' || !('__NEXT_DATA__' in window))
+    return;
+
+  // @ts-expect-error: _isNextRouter is only set on the Next.js router
+  const isUsingNextRouter = routing !== true && routing?.router?._isNextRouter;
+
+  warn(
+    isUsingNextRouter,
```

Comment:
I believe if this whole condition is inlined, it will not be in the prod bundle, and thus also not needed to increase bundlesize

Reply:
I still wrapped the whole thing into an IIFE for readability.
It still is a external function too cause I don't wanna pollute the hook itself